content/exchange/artifacts/Generic.Remediation.UploadFile.Glob.yaml

```diff
@@ -0,0 +1,32 @@
+name: Generic.Remediation.UploadFile.Glob
+description: |
+   This artefact lets you upload a file to the target system. If the target file exists, it will be overridden.
+   It uses glob to define the destination path. You can upload the file to multiple destinations at once.
+
+type: CLIENT
+
+parameters:
+   - name: TargetGlob
+     description: Destination Path, if the destination already exists, it will be overridden
+   - name: ReallyDoIt
+     description: When selected, it will really override the targets!
+     type: bool
+     
+tools:
+  - name: RemediationFile
+    url:
+
+sources:
+  - query: |
+      LET targets = SELECT * FROM glob(globs=TargetGlob)
+        WHERE NOT IsDir
+        ORDER BY OSPath DESC 
+     
+      LET UploadedFile <= SELECT FullPath FROM Artifact.Generic.Utils.FetchBinary(ToolName="RemediationFile", IsExecutable=FALSE, TemporaryOnly=TRUE)
+     
+      LET upload_targets = SELECT *, copy(filename=UploadedFile.FullPath[0], accessor="file", dest=OSPath) as Overridden FROM targets
+
+      SELECT OSPath,Overridden,Size,Mtime,Ctime,Btime,IsDir,IsLink
+      FROM if(condition=ReallyDoIt,
+            then= upload_targets,
+            else= { SELECT *, FALSE as Overridden FROM upload_targets } )
```

Comment:
Have you tested this actually stops it from overriding? I think it might be safer to just use an if() here:

```
if(condition = ReallyDoIt, then=copy(filename=UploadedFile.FullPath[0], accessor="file", dest=OSPath)) AS Overwritten
```

Reply:
Yes it works fine. I used it multiple times during my last engagement. Most of the logic is copied from another artefact: https://docs.velociraptor.app/exchange/artifacts/pages/globremediation/

Reply:
Its not really the same as that artifact - you can see in GlobRemediation the bottom query is

```
SELECT ...
      FROM if(condition=ReallyDoIt,
            then= delete_targets,
            else= { SELECT *, FALSE as Removed FROM targets } )
```
So depending on the condition it will choose a completely different query - one for delete and one for just showing.

But in this artifact it chooses the same query in both cases, it just replaces the column that removes with FALSE. This generally also works because VQL is generally lazy and wont evaluate a column that got replaced but it is quite fragile and potentially dangerous in this case.

Reply:
Thanks for your explanations. I've changed it to an if statement and simplified the query a bit.